library/src/meta/footnote.rs

```diff
@@ -58,9 +87,64 @@ pub struct FootnoteElem {
     #[default(Numbering::Pattern(NumberingPattern::from_str("1").unwrap()))]
     pub numbering: Numbering,
 
-    /// The content to put into the footnote.
+    /// The content to put into the footnote. Can also be the label of another
+    /// footnote this one should point to.
     #[required]
-    pub body: Content,
+    pub body: FootnoteBody,
```

Comment:
Should this still be called "body"? This term seems missleading. I can't think of a better alternative though. Maybe "value"?

Reply:
See how `text` does it with font size and fill:
https://github.com/typst/typst/blob/c0c03812d08bc833b8a2477c594d762f43a0c95b/library/src/text/mod.rs#L167-L180

Something similar should be done (imho): have a `body` (content-only) parameter separate from some sort of `referee` (label) parameter, and have both be optional. You could then add a check in the Show code to bail if neither the body or the label were specified.

Reply:
I think allowing a label as body isn't that bad and simpler.

Reply:
I'll leave it that way for now, then.